### DISCUSSION TITLE
'charon-svc' has unmet dependency: CUSTOM:socket when launching on Windows

### DISCUSSION THREAD

**Discussion by kcchenkd** [strongswan/strongswan#1204] [Q&A]
Hi,
I am trying to cross-compile strongSwan 5.9.7 for Windows 10 Home using MinGW64 on Ubuntu 16.04.2 LTS

I am able to run the ./configure successfully with following parameters, and run "make" and "make install DESTDIR=/home/eric/dist_3/" comamnds successfully
CFLAGS="-g -O2 -Wall -Wno-pointer-sign -Wno-format-security -Wno-format -mno-ms-bitfields -I/home/eric/OpenSSL_1_1_1-stable/build/include" \
LDFLAGS="-L/home/eric/OpenSSL_1_1_1-stable/build/lib" \
./configure --disable-defaults --enable-monolithic --enable-static --enable-svc --enable-ikev2 --enable-ikev1  --enable-nonce \
 --enable-pem --enable-pkcs1 --enable-x509 --enable-openssl --enable-socket-win  --enable-kernel-wfp \
 --enable-kernel-iph --enable-pubkey --enable-swanctl --enable-python-eggs \
 --with-swanctldir=swanctl --with-strongswan-conf=strongswan.conf --host=x86_64-w64-mingw32

Then I copy binaries and libs to Windows, and see the following errors when launching charon-svc.exe
C:\strongswan\dist_3\usr\local\bin>charon-svc.exe
00[DMN] Starting IKE service charon-svc (strongSwan 5.9.7, Windows Client 6.2.9200 (SP 0.0)
00[KNL] registering WFP provider failed: 0x00000005
00[LIB] feature CUSTOM:libcharon-receiver in critical plugin 'charon-svc' has unmet dependency: CUSTOM:socket
00[LIB] feature CUSTOM:libcharon in critical plugin 'charon-svc' has unmet dependency: CUSTOM:libcharon-receiver
00[LIB] failed to load 2 critical plugin features

I have tried the Windows Native Build option as well, and seeing same error.

My understanding is libcharon should try to use socket-win plugin instead of socket-default on Windows. I confirmed that socket-win plugin is enabled. Is the error indicates that libcharon could not load socket-win plugin or it tries to load socket-default? 

Any one has any idea what is missing? 

Thanks in advance!

**Comment by Thermi** [comment]
`00[KNL] registering WFP provider failed: 0x00000005`

I guess that's the first problem you should fix.

**Reply by tobiasbrunner** [reply]
Probably caused if the IKEEXT service isn't stopped/disabled (see [here](https://docs.strongswan.org/docs/5.9/os/windows.html)).

**Comment by wsm25** [comment]
I suppose you are not running it with admin privileges.
